packages/embark-ui/src/containers/ServicesContainer.js

```diff
@@ -21,7 +23,15 @@ class ServicesContainer extends Component {
   }
 
   render() {
-    return <DataWrapper shouldRender={this.props.services.length > 0 } {...this.props} render={({services}) => (
+    const {error, loading, services} = this.props;
+    if (error) {
+      return <Error error={error} />;
+    }
+
+    if (loading) {
+      return <Loading />;
+    }
+    return <DataWrapper shouldRender={services.length > 0 } {...this.props} render={({services}) => (
```

Comment:
Isn't the DataWrapper supposed to handle the loading for us, or is there another loading that needs to be taken into account?

Reply:
The `DataWrapper` was handling the loading for us, however the `DataWrapper` was causing infinite `ContractsContainer` reloads once a network error was received (after the API was killed via `service api off`), therefore it had to be removed. I tried quite a lot of different solutions to keep the `DataWrapper`, including making it a HOC, however, nothing seemed to prevent the infinite reload issue.

Reply:
But the DataWrapper is still here though? Maybe we should remove it here, because we handle the loading and error ourselves?

Reply:
True, sorry that must have been a brain fart! They were definitely not needed changes and have been reverted.